### ISSUE TITLE
Set image expiry

### ISSUE THREAD

**Issue by alltheseas** [damus-io/damus#1565]
With the exception of PFPs, set an image auto-expiry date for images, and videos to control damus cache size. 

Related: fix clear cache https://github.com/damus-io/damus/issues/1472

Explore storage visualization, preferences https://github.com/damus-io/damus/issues/1554

**Comment by danieldaquino** [comment]
Investigating this now!

**Comment by danieldaquino** [comment]
I created a patch to address this!

I set:
- 1 week for images on notes
- 2 weeks for profile banners
- Never for profile pictures

All code, testing, and discussion was sent via email, and can be viewed here: https://groups.google.com/a/damus.io/g/patches/c/cGZcTD8zYLE

**Comment by alltheseas** [comment]
> I created a patch to address this!
> 
> I set:
> - 1 week for images on notes
> - 2 weeks for profile banners
> - Never for profile pictures
> 
> All code, testing, and discussion was sent via email, and can be viewed here: https://groups.google.com/a/damus.io/g/patches/c/cGZcTD8zYLE

Nice!

Does it include videos as well?

**Comment by danieldaquino** [comment]
Thanks @alltheseas, that's a good point. The patch does not currently include videos.

However, I examined the code, and it does not seem like videos are cached at all. 🤔

I already looked at:
- `DamusViewPlayer.swift`
- `DamusVideoPlayerViewModel.swift`
- `AVPlayer`
- `ImageCarousel.swift`

In #1472 I remember I did not see any evidence that videos are being cached either, even though someone mentioned that they are.

@jb55, would you happen to know if we cache videos and — if we do — in which part of the code we do that? Am I looking at the right places?

**Comment by jb55** [comment]
On Fri, Oct 13, 2023 at 05:45:45PM -0700, Daniel D’Aquino wrote:
>I created a patch to address this!
>
>I set:
>- 1 week for images on notes
>- 2 weeks for profile banners
>- Never for profile pictures

this is great. good job man.

**Comment by jb55** [comment]
On Fri, Oct 13, 2023 at 06:16:21PM -0700, Daniel D’Aquino wrote:
>Thanks @alltheseas, that's a good point. The patch does not currently include videos.
>
>However, I examined the code, and it does not seem like videos are cached at all. 🤔
>
>I already looked at:
>- `DamusViewPlayer.swift`
>- `DamusVideoPlayerViewModel.swift`
>- `AVPlayer`
>- `ImageCarousel.swift`
>
>In #1472 I remember I did not see any evidence that videos are being cached either, even though someone mentioned that they are.
>
>@jb55, would you happen to know if we cache videos and — if we do — in which part of the code we do that? Am I looking at the right places?

that's really bad if it were the case. I assumed the video player would
cache things itself, but if not we'll definitely need to fix that. maybe
check the container storage if you find anything?

**Comment by danieldaquino** [comment]
> maybe check the container storage if you find anything?

I checked the container of the normal Damus instance running on my phone (4GB+ of data) and could not find any videos. Here is what I tried:

1. Ran `find .` at the root of the container and read all file paths. Nothing stood out as being videos.
2. Ran `file *` on some folders that looked like caches and manually inspected output. No video files detected
3. Ran `find . -type f -exec file {} \; | grep video` at the root of the container to see if it detects any files with video content. Output was blank
4. Ran `find . -type f -exec file -b {} \;` at the root of the container and skim through output. Saw a lot of images, some other file formats, but no videos
